tests/unit/test_column_noise.py

```diff
@@ -161,109 +156,112 @@ def test_miswrite_ages(dummy_dataset, default_configuration):
-            "row_noise_level": 1,
-            "possible_perturbations": perturbations,
-            "possible_perturbation_levels": probabilities,
+            "decennial_census": {
+                "age": {
+                    "age_miswriting": {
+                        "row_noise_level": 1,
+                        "possible_perturbations": perturbations,
+                    },
+                },
+            },
         },
-    )
+    )["decennial_census"]["age"]["age_miswriting"]
 
-    data = pd.Series([str(age)] * num_rows)
+    data = pd.Series([str(original_age)] * num_rows)
     noised_data = NOISE_TYPES.AGE_MISWRITING(data, config, RANDOMNESS0, "test")
-    for i in range(len(perturbations)):
-        expected_noise = probabilities[i]
-        perturbed_age = str(age + perturbations[i])
-        actual_noise = (noised_data == perturbed_age).mean()
+    for perturbation in perturbations:
+        expected_noise = perturbations[perturbation]
+        actual_noise = (noised_data.astype(int) - original_age == perturbation).mean()
         assert np.isclose(actual_noise, expected_noise, rtol=0.01)
 
 
-def test_miswrite_ages_handles_perturbation_to_same_age(default_configuration):
+def test_miswrite_ages_handles_perturbation_to_same_age():
     """Tests an edge case. It's possible that after an age is perturbed it ends
     up being the original age. In that case, subtract 1. eg, an age of 1 that is
-    perturbed becomes -1. But we cannot have negative so we flip the sign to +1.
+    perturbed -2 becomes -1. But we cannot have negative so we flip the sign to +1.
     But that's the same as the original age and so should become 1-1=0.
     """
     num_rows = 100
     age = 1
-    perturbations = [-1]
+    perturbations = [-2]  # This will cause -1 which will be flipped to +1
 
-    config = default_configuration["decennial_census"]["age"]["age_miswriting"]
-    config.update(
+    config = get_configuration(
         {
-            "row_noise_level": 1,
-            "possible_perturbations": perturbations,
+            "decennial_census": {
+                "age": {
+                    "age_miswriting": {
+                        "row_noise_level": 1,
+                        "possible_perturbations": perturbations,
+                    },
+                },
+            },
         },
-    )
+    )["decennial_census"]["age"]["age_miswriting"]
 
     data = pd.Series([str(age)] * num_rows)
     noised_data = NOISE_TYPES.AGE_MISWRITING(data, config, RANDOMNESS0, "test")
 
     assert (noised_data == "0").all()
 
 
-def test_miswrite_ages_fails_if_bad_probs(default_configuration):
-    """Test that runtimerrors if probs do not add up to 1"""
-    probabilities = [0.1, 0.8]  # does not sum to 1
+def test_miswrite_ages_flips_negative_to_positive():
+    """Test that any ages perturbed to <0 are reflected to positive values"""
+    num_rows = 100
+    age = 3
+    perturbations = [-5]  # This will cause -2 and should flip to +2
```

Comment:
Since we know that a perturbation of -6 would also send this to +2, I think -4 or -7 would be better values to go with an original age of 3.